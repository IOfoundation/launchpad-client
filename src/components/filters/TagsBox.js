import React, {PureComponent} from 'react';
import {PropTypes} from 'prop-types';
import {isEmpty, isString} from 'lodash';
import Chip from '../shared/Chip';

class TagsBox extends PureComponent {
  render() {
    const filters = this.props.filters;
    return (
      <div>
        {isString(filters.category) && (
          <Chip
            key={filters.category}
            text={filters.category}
            handleClick={this.props.deleteFilter}
            canDelete={true}
          />
        )}
        {!isEmpty(filters.category) && !isString(filters.category) ?
          filters.category.map(filter => (
            <Chip
              key={filter}
              text={filter}
              handleClick={this.props.deleteFilter}
              canDelete={true}
            />
          ))
        : ''}
        {filters.category && (
          <a
            className="search-filter-clear text-thin"
            onClick={this.props.clearAll}
          >
            <span>{'Clear All'}</span>
          </a>
        )}
      </div>
    );
  }
}
TagsBox.propTypes = {
  clearAll: PropTypes.func.isRequired,
  deleteFilter: PropTypes.func.isRequired,
  filters: PropTypes.object.isRequired,
};

export default TagsBox;
