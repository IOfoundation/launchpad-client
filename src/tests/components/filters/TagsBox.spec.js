import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import TagsBox from 'components/filters/TagsBox';

const createProps = props => {
  return {
    clearAll: jest.fn(),
    deleteFilter: jest.fn(),
    filters: props.filters,
  };
};

describe('<TagsBox />', () => {
  it('renders snapshot of TagsBox', () => {
    const props = createProps({
      filters: {category: []},
    });
    const wrapper = shallow(<TagsBox {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
