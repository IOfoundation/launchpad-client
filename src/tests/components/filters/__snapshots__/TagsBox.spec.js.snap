// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagsBox /> renders snapshot of TagsBox 1`] = `<div />`;
